### PR TITLE
SIGINT-2057, SIGINT-2106:  expose SRM parameters and additional SRM Parameters to be supported

### PR DESCRIPTION
**Pull request by siyaramsynopsys**

### CODE REVIEW

File: synopsys-task/src/synopsys-task/application-constant.ts [hunk 1]
```diff
@@ -197,6 +198,11 @@ export const BRIDGE_COVERITY_INSTALL_DIRECTORY_KEY =
 export const COVERITY_INSTALL_DIRECTORY_KEY = "coverity_install_directory";
 export const COVERITY_INSTALL_DIRECTORY_KEY_CLASSIC_EDITOR =
   "bridge_coverity_install_directory";
+
+export const COVERITY_EXECUTION_PATH_KEY = "coverity_execution_path";
+export const COVERITY_EXECUTION_PATH_KEY_CLASSIC_EDITOR =
+  "bridge_coverity_execution_path";
```
Comment (maksudur-rahman-maruf) [inline synopsys-task/src/synopsys-task/application-constant.ts:204]: For classic editor we are no longer following the snake_case convention.
 Now we are following the camelCase convention. Can you apply the camelCase for all the new fields?

File: synopsys-task/src/synopsys-task/input.ts [hunk 2]
```diff
@@ -138,6 +142,20 @@ function getInputForPolarisAssessmentMode() {
       : "")
   );
 }
+function getInputForSrmAssessmentMode() {
```
Comment (maksudur-rahman-maruf) [inline synopsys-task/src/synopsys-task/input.ts:145]: I believe the assessment mode is not specified in the SRM requirements. Can you please remove the SRM assessment mode-related code?

File: synopsys-task/test/unit/synopsys-task/tools-parameter.spec.ts [hunk 3]
```diff
@@ -1358,4 +1358,225 @@ describe("Synopsys Tools Parameter test", () => {
...
+            try {
+                const formattedCommand = await synopsysToolsParameter.getFormattedCommandForSrm();
+            } catch (e) {
+                const errorObj = e as Error;
+                expect(errorObj.message).contains('Invalid value for '.concat(constants.SRM_ASSESSMENT_TYPES_KEY))
+                expect(errorObj.message).contains(ErrorCode.INVALID_SRM_ASSESSMENT_TYPES.toString())
+            }
+        });
+
+        it('should success for SRM command formation with mandatory parameters', async function () {
```
Comment (maksudur-rahman-maruf) [inline synopsys-task/test/unit/synopsys-task/tools-parameter.spec.ts:1407]: Please add a test case for defaulting the project name.

Comment (maksudur-rahman-maruf) [inline synopsys-task/test/unit/synopsys-task/tools-parameter.spec.ts:1407]: Please add a test case to ensure the project name defaults correctly

File: synopsys-task/src/synopsys-task/application-constant.ts [hunk 4]
```diff
@@ -546,29 +546,27 @@ export const BLACKDUCK_ARGS_KEY_CLASSIC_EDITOR_FOR_SRM = "blackduckArgsForSrm";
 
 //SRM
 export const SRM_URL_KEY = "srm_url";
-export const SRM_URL_KEY_CLASSIC_EDITOR = "bridge_srm_url";
+export const SRM_URL_KEY_CLASSIC_EDITOR = "bridgeSrmUrl";
```
Comment (maksudur-rahman-maruf) [inline synopsys-task/src/synopsys-task/application-constant.ts:549]: remove bridge from all the new fields

File: synopsys-task/src/synopsys-task/model/srm.ts [hunk 5]
```diff
@@ -12,7 +12,7 @@ export interface Srm {
 export interface SrmData {
   url: string;
   apikey: string;
-  assessment: { types: string[]; mode?: string };
+  assessment: { types: string[] };
   project?: { name?: string; id?: string };
   branch?: BranchInfo;
   azure?: AzureData;
```
Comment (maksudur-rahman-maruf) [inline synopsys-task/src/synopsys-task/model/srm.ts:18]: I think we do not need azure here.

**Review by maksudur-rahman-maruf — APPROVED** [review]

File: synopsys-task/task.json [hunk 6]
```diff
@@ -97,7 +115,8 @@
         "": "",
         "polaris": "Polaris",
         "blackduck": "Black Duck",
-        "coverity": "Coverity Platform"
+        "coverity": "Coverity Platform",
+        "srm": "Software Risk Manager"
```
Comment (maksudur-rahman-maruf) [inline synopsys-task/task.json:119]: Software Risk Manager -> Software Risk Manager (SRM). Please apply this change

File: synopsys-task/src/synopsys-task/validator.ts [hunk 7]
```diff
@@ -130,3 +131,20 @@ export function validateBlackDuckInputs(): string[] {
...
+      constants.SRM_ASSESSMENT_TYPES_KEY,
+      inputs.SRM_ASSESSMENT_TYPES
+    );
+    errors = validateParameters(paramsMap, constants.SRM_KEY);
+  }
+  return errors;
+}
+export function isNullOrEmptyValue(param: string): boolean {
+  return param == null || param.length === 0;
+}
```
Comment (zaman-akib) [inline synopsys-task/src/synopsys-task/validator.ts:150]: This method can be removed since it has no usage.

File: synopsys-task/task.json [hunk 8]
```diff
@@ -3,7 +3,7 @@
   "id": "edd0f606-d9e3-4294-9d0d-64f9f1aadfde",
   "name": "SynopsysSecurityScan",
   "friendlyName": "Synopsys Security Scan",
-  "description": "Run security scans on Polaris, Black Duck or Coverity platforms",
+  "description": "Run security scans on Polaris, Black Duck, Coverity or Software Risk Manager platforms",
```
Comment (zaman-akib) [inline synopsys-task/task.json:6]: Add SRM in brackets like  Software Risk Manager (SRM)

File: vss-extension-dev.json [hunk 9]
```diff
@@ -19,7 +19,7 @@
     "Open Source",
     "Build"
   ],
-  "description": "Run security scans on Polaris, Black Duck or Coverity platforms",
+  "description": "Run security scans on Polaris, Black Duck, Coverity or Software Risk Manager platforms",
```
Comment (zaman-akib) [inline vss-extension-dev.json:22]: Add SRM in brackets like Software Risk Manager (SRM)

File: vss-extension.json [hunk 10]
```diff
@@ -20,7 +20,7 @@
     "Open Source",
     "Build"
   ],
-  "description": "Run security scans on Polaris, Black Duck or Coverity platforms",
+  "description": "Run security scans on Polaris, Black Duck , Coverity or Software Risk Manager platforms",
```
Comment (zaman-akib) [inline vss-extension.json:23]: Add SRM in brackets like Software Risk Manager (SRM)

File: synopsys-task/src/synopsys-task/tools-parameter.ts [hunk 11]
```diff
@@ -644,6 +648,116 @@ export class SynopsysToolsParameter {
...
+    const blackduckArgs = this.setBlackDuckArbitraryArgs();
+
+    if (Object.keys(coverityArgs).length > 0) {
+      srmData.data.coverity = { ...srmData.data.coverity, ...coverityArgs };
+    }
+    if (Object.keys(blackduckArgs).length > 0) {
+      srmData.data.blackduck = { ...srmData.data.blackduck, ...blackduckArgs };
+    }
+
+    const inputJson = JSON.stringify(srmData);
```
Comment (zaman-akib) [inline synopsys-task/src/synopsys-task/tools-parameter.ts:736]: to filter the empty data use the method `filterEmptyData()` before preparing json.

```
// Remove empty data from json object
srmData = filterEmptyData(srmdata);
```

**Review by zaman-akib — CHANGES_REQUESTED** [review]

**Review by zaman-akib — APPROVED** [review]

File: synopsys-task/src/synopsys-task/model/srm.ts [hunk 12]
```diff
@@ -0,0 +1,32 @@
+import { CoverityArbitrary } from "./coverity";
+import { BlackDuckArbitrary } from "./blackduck";
+
+export interface Srm {
+  srm: SrmData;
+  coverity?: coverityDetails;
```
Comment (spurohitsynopsys) [inline synopsys-task/src/synopsys-task/model/srm.ts:6]: Can you rename coverityDetails to CoverityDetails to follow consistent naming conventions across all.

File: synopsys-task/src/synopsys-task/synopsys-bridge.ts [hunk 13]
```diff
@@ -192,10 +201,29 @@ export class SynopsysBridge {
...
     }
     return [formattedCommand, errors];
   }
 
+  private async prepareSrmCommand(
+    formattedCommand: string,
+    tempDir: string
+  ): Promise<[string, string[]]> {
+    const srmErrors: string[] = validateSrmInputs();
+    const commandFormatter = new SynopsysToolsParameter(tempDir);
```
Comment (spurohitsynopsys) [inline synopsys-task/src/synopsys-task/synopsys-bridge.ts:218]: maybe we can move` const commandFormatter = new SynopsysToolsParameter(tempDir);` into if block after L219

**Review by spurohitsynopsys — APPROVED** [review]

**Pull request merged**